### ISSUE TITLE
version 5-dev: make testall stops at examples

### ISSUE THREAD

**Issue by colbec** [JuliaLang/julia#14136]
Using opensuse Leap 42.1, from compile of latest clone of Julia from git.
Compile seems to run correctly.
Make testall goes well until the last entry "examples", where the process seems to hang.
I think examples is ok, since "make test-examples" passes.
At this point examination of the computer resources shows that the julia processes are all at 0% CPU.
No further output after 5 minutes. Suggestions?

On a second issue, from time to time I get a pop up window from opensuse Yast saying that something has to run as root. If I accept this, there is nothing in the following installation window. From time to time I see a reference to "apt-*" which of course this system will not have.

**Comment by tkelman** [comment]
> Make testall goes well until the last entry "examples", where the process seems to hang.

Can you record the entire output of `make testall` and upload a gist with the log? Most likely some earlier test is failing or freezing, and the rest of the workers are waiting at the end for that worker to finish.

> On a second issue, from time to time I get a pop up window from opensuse Yast saying that something has to run as root.

When exactly does this occur? If it's while installing packages, that is a BinDeps / individual package issue since we don't yet have a `zypper` provider for opensuse. It would likely be simple to add one though.

**Comment by colbec** [comment]
Here is the output from my "make testall";  execution was interrupted with a control C (^c) in the log. 
[testall.txt](https://github.com/JuliaLang/julia/files/44407/testall.txt)

The issue with Yast happens for example when I start up the IDE with "ipython notebook --profile=julia".
As you say the recompile process gets to the bin deps and then calls yast. Once the recompile of the components is done, the IDE seems to work fine, including PyPlot.

At one point I thought it also happened with the Nettle package when doing Pkg.add(), then I found that I did not have the devel packages from Yast for Nettle loaded into the system. I corrected that and things seemed to go better.

**Comment by mauro3** [comment]
I see the same on Arch Linux: https://gist.github.com/mauro3/d5a3579f0ff8e13f8502  

It looks like @tkelman is right, when running `make test-socket` by itself it hangs.

Also, probably unrelated, the printing is somewhat messed-up: the test totals get printed after libgit2.

**Comment by tkelman** [comment]
libgit2 is using some of the new testset features which print differently, should probably be turned back to normal `@test` to fit with the printing of the rest of the base tests.

**Comment by colbec** [comment]
Using Julia Version 0.5.0-dev+1534 (2015-12-01 06:23 UTC) on openSUSE Leap 42.1:
The tests now complete with SUCCESS message.
The last test relates to compile, which generates two Loaderror messages annotated as normal.
FYI and thankyou.

**Comment by tkelman** [comment]
Should we close this issue then? May have been fixed by #14133?

**Comment by mauro3** [comment]
Works for me too. Closing is fine by me.

**Comment by colbec** [comment]
The only thing is that (possibly in error) I spoke of two issues, the second being these odd calls that bring up Yast for no apparent reason. It is still happening, for example in a call to PyPlot package. It would be fine by me to close this issue, but I would take this as a hint that a separate issue was needed for the zypper case, and I will do that. I am struggling to find more information that might help lead to a resolution of this as well.

**Comment by tkelman** [comment]
Those are almost certainly issues with individual packages, not base Julia. See https://github.com/JuliaLang/BinDeps.jl for the code that most packages use to install binary dependencies. There were some past pull requests there to add support for pacman on Arch, and there's existing code for apt-get and yum support. Adding support for zypper should first be done there in terms of how do you call zypper for different commands, then individual packages should have `provides` lines added to their `build/deps.jl` files listing the OpenSUSE package names for zypper.
